src/core/component.test.ts

```diff
@@ -713,6 +713,31 @@ describe("nested option", () => {
         });
     });
 
+    it("watches should be initialized once", (done) => {
+        const vm = new Vue({
+            template:
+                `<test-component>` +
+                `  <nested :prop1="value" />` +
+                `</test-component>`,
+            components: {
+                TestComponent,
+                Nested
+            },
+            props: ["value"],
+            propsData: {
+                value: 123
+            }
+        }).$mount();
+
+        vm.$props.value = 456;
+
+        Vue.nextTick(() => {
+            const nestedConfig = (vm.$children[0] as any ).$children[0].$vnode;
+            expect(nestedConfig.componentInstance._watchers.length).toEqual(3);
```

Comment:
Why 3 watchers are expected? How does this get along with the fact that the `Nested` component has 2 props?

Reply:
Maybe modify it in this way:
```ts
        const nestedConfig = (vm.$children[0] as any ).$children[0].$vnode;
        const expected = nestedConfig.componentInstance._watchers.length;

	vm.$props.value = 456;

        Vue.nextTick(() => {
            expect(nestedConfig.componentInstance._watchers.length).toEqual(expected);
            done();
        });
```

Reply:
vue create watcher for component. You can see it [here](https://codesandbox.io/s/cool-edison-w27ep)